config/main.go

```diff
@@ -37,3 +44,93 @@ var (
 	// if RequirePass is set to an empty string, no authentication is required
 	RequirePass string = constants.EmptyStr
 )
+
+func init() {
+	args := os.Args[1:]
+
+	if len(args) == 1 && args[0] == constants.InitConfig {
+
+		err := os.MkdirAll(filepath.Dir(ConFilePath), 0755)
+		if err != nil {
+			if os.IsPermission(err) {
+				log.Error("Permission denied to create directory")
+			}
+			log.Errorf("Failed to create directory: %v", err)
+		}
+
+		if _, err := os.Stat(ConFilePath); err == nil {
+			log.Infof("Config file already exists at %s", ConFilePath)
+			ParseConfig(ConFilePath)
+			return
+		}
+
+		configFile, err := os.OpenFile(ConFilePath, os.O_CREATE|os.O_RDWR, 0666)
+		if err != nil {
+			log.Fatalf("Failed to create config file: %v", err.Error())
```

Comment:
Will not able to create a config file be a fatal error? Or shall we use defaults in that case?
cc: @JyotinderSingh thoughts?

Reply:
It can be a warning, and we should proceed with defaults.